Add support for compression_codec to logging file sink endpoints

kellymclaughlin opened the pull request:
This follows from [this](https://github.com/fastly/go-fastly/pull/235) `go-fastly` PR.

Integralist commented on pkg/app/run_test.go:
```diff
@@ -790,6 +790,16 @@ COMMANDS
         --server-side-encryption-kms-key-id=SERVER-SIDE-ENCRYPTION-KMS-KEY-ID
```
I went through each documented logging endpoint checking which one's supported `compression_codec` and from what I can see is that the following log endpoints support it but aren't included in this PR. 

Is there a reason for this absence? 

- Cloudfiles
- DigitalOcean
- Kafka

**UPDATE**: I can see that Kafka appears to have slightly different compression values, which maybe why you've not included that one in this PR, but the other two (Cloudfiles and DigitalOcean appear to have the same values as what's defined in this PR).

kellymclaughlin replied:
Good question. Kafka already has the compression support. Cloudfiles and DigitalOcean aren't included only because they weren't on the original list I had of our endpoints that were part of this addition. I'll need to add the compression support for these two to `go-fastly` before I can include them in this cli change.

Integralist replied:
👍🏻

Integralist reviewed (commented):
Thanks @kellymclaughlin 

I only have a few comments that require clarification/addressing.

kellymclaughlin commented on pkg/logging/azureblob/azureblob_integration_test.go:
```diff
@@ -272,13 +272,13 @@ func createBlobStorageOK(i *fastly.CreateBlobStorageInput) (*fastly.BlobStorage,
 		MessageType:       "classic",
 		Placement:         "none",
 		ResponseCondition: "Prevent default logging",
+		CompressionCodec:  "zstd",
```
There was indeed a discrepancy between the docs and the values accepted by the API for this. The docs are now updated to use the value `zstd`.

Integralist commented on pkg/app/run_test.go:
```diff
@@ -790,6 +790,16 @@ COMMANDS
                                  server-side-encryption is set to aws:kms
+        --compression-codec=COMPRESSION-CODEC
+                                 The codec used for compression of your logs.
+                                 Valid values are zstandard, snappy, and gzip.
```
Let's update `zstandard` to be `zstd` (across this entire file) as per the latest update to the API reference.

kellymclaughlin replied:
Addressed by https://github.com/fastly/cli/pull/190/commits/fda2657dc6aa05f1570de57996c68b73f62aadc3

Integralist commented:
@kellymclaughlin I wanted to just quickly circle back round to this PR to make sure you weren't waiting on me for anything else (e.g. as far as I'm aware this PR is still with you to address the various comments I left in my last review). If so all good 👍🏻 otherwise if you do need anything from me, then just let me know. Thanks!

Integralist commented:
👋🏻 @kellymclaughlin I can see there are still some comments from my original review that are still outstanding. If this is a low priority PR for you then just let me know and I'll stop pinging you reminders about it 🙂

kellymclaughlin commented:
> 👋🏻 @kellymclaughlin I can see there are still some comments from my original review that are still outstanding. If this is a low priority PR for you then just let me know and I'll stop pinging you reminders about it

I had to ignore this for most of last week due to some other work, but I'm going to be addressing the comments this week. I'll start with a new `go-fastly` PR to add the requisite changes for the cloudfiles and digitialocean endpoints and then I'll incorporate the changes for those two endpoints into this PR as well. In the meantime I'll address the other comments as well. Thanks!

Integralist commented:
Awesome. I'll keep an eye out for the go-fastly PR 👍🏻

Integralist commented:
@kellymclaughlin might be worth rebasing the main branch and get this over the finish line 🙂

kellymclaughlin commented:
This is still blocked on my forthcoming `go-fastly` PR which is in turn waiting on an API fix. I can rebase and make sure it's tracking the main line though.

kellymclaughlin commented on pkg/logging/azureblob/azureblob_integration_test.go:
```diff
@@ -317,14 +317,14 @@ func listBlobStoragesOK(i *fastly.ListBlobStoragesInput) ([]*fastly.BlobStorage,
 			SASToken:          "token",
 			Path:              "/logs",
 			Period:            86400,
-			GzipLevel:         9,
```
Addressed by https://github.com/fastly/cli/pull/190/commits/fda2657dc6aa05f1570de57996c68b73f62aadc3

kellymclaughlin commented on pkg/logging/azureblob/azureblob_test.go:
```diff
@@ -163,14 +166,15 @@ func createCommandAll() *CreateCommand {
 		Path:              common.OptionalString{Optional: common.Optional{WasSet: true}, Value: "/log"},
 		Period:            common.OptionalUint{Optional: common.Optional{WasSet: true}, Value: 3600},
-		GzipLevel:         common.OptionalUint{Optional: common.Optional{WasSet: true}, Value: 2},
+		GzipLevel:         common.OptionalUint{Optional: common.Optional{WasSet: false}, Value: 2},
```
Addressed by https://github.com/fastly/cli/pull/190/commits/fda2657dc6aa05f1570de57996c68b73f62aadc3

kellymclaughlin commented on pkg/logging/azureblob/azureblob_test.go:
```diff
@@ -201,14 +205,15 @@ func updateCommandAll() *UpdateCommand {
 		Path:              common.OptionalString{Optional: common.Optional{WasSet: true}, Value: "new5"},
 		Period:            common.OptionalUint{Optional: common.Optional{WasSet: true}, Value: 3601},
-		GzipLevel:         common.OptionalUint{Optional: common.Optional{WasSet: true}, Value: 3},
+		GzipLevel:         common.OptionalUint{Optional: common.Optional{WasSet: false}, Value: 3},
```
Addressed by https://github.com/fastly/cli/pull/190/commits/fda2657dc6aa05f1570de57996c68b73f62aadc3

kellymclaughlin commented on pkg/logging/ftp/ftp_test.go:
```diff
@@ -162,12 +164,13 @@ func createCommandAll() *CreateCommand {
 		Path:              common.OptionalString{Optional: common.Optional{WasSet: true}, Value: "/logs"},
 		Period:            common.OptionalUint{Optional: common.Optional{WasSet: true}, Value: 3600},
-		GzipLevel:         common.OptionalUint8{Optional: common.Optional{WasSet: true}, Value: 2},
+		GzipLevel:         common.OptionalUint8{Optional: common.Optional{WasSet: false}, Value: 2},
```
Addressed by https://github.com/fastly/cli/pull/190/commits/fda2657dc6aa05f1570de57996c68b73f62aadc3

kellymclaughlin commented:
@Integralist I'm not sure why it's taking the tests so long to run, but the failed test is due to a timeout. I don't think I have the proper permissions to re-run the workflow to try it again. If you do maybe you could give it another go and see if it makes it through.

Integralist commented:
@kellymclaughlin is this ready for review?

Integralist reviewed (approved):
@kellymclaughlin LGTM. Let me know if you're happy for me to merge this?

kellymclaughlin commented:
> @kellymclaughlin LGTM. Let me know if you're happy for me to merge this?

Thanks for the review! Yep, this is ready to merge.
